tests/fault_tolerance/test_request_cancellation_trtllm.py

```diff
@@ -0,0 +1,389 @@
+                except Exception as e:
+                    pytest.fail(f"Stream reading failed: {e}")
+
+            response.close()
+            raise Exception("Closed response")
+        else:
+            pytest.fail(f"Unknown request type: {request_type}")
+
+        pytest.fail(
+            f"{request_type} request completed unexpectedly - should have been cancelled"
+        )
+    except Exception as e:
+        logger.info(f"{request_type} request was cancelled: {e}")
+
+
+def read_log_content(log_path: str | None) -> str:
+    """Read log content from a file"""
+    if log_path is None:
+        pytest.fail("Log path is None - cannot read log content")
+
+    try:
+        with open(log_path, "r") as f:
+            return f.read()
+    except Exception as e:
+        pytest.fail(f"Could not read log file {log_path}: {e}")
+
+
+def strip_ansi_codes(text: str) -> str:
+    """Remove ANSI color codes from text"""
+    ansi_escape = re.compile(r"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])")
+    return ansi_escape.sub("", text)
+
+
+def verify_request_cancelled_trtllm_simple(
+    trtllm_process: TRTLLMProcess,
+    trtllm_log_offset: int = 0,
+) -> int:
+    """Simplified verification for TRTLLM cancellation messages
+
+    Returns:
+        int: new_trtllm_log_length
+    """
+
+    # Check TRTLLM log for cancellation pattern
+    trtllm_log_content = read_log_content(trtllm_process._log_path)
+    new_trtllm_content = trtllm_log_content[trtllm_log_offset:]
+
+    # Look for TRTLLM-specific cancellation pattern
+    has_trtllm_cancellation = False
+    cancellation_pattern = "Aborted Request ID: "
+
+    for line in new_trtllm_content.split("\n"):
+        clean_line = strip_ansi_codes(line).strip()
+        if cancellation_pattern in clean_line:
+            has_trtllm_cancellation = True
+            logger.info(f"Found TRTLLM cancellation message: {clean_line}")
+            break
+
+    if not has_trtllm_cancellation:
+        pytest.fail(f"Could not find '{cancellation_pattern}' in TRTLLM backend log")
+
+    return len(trtllm_log_content)
+
+
+@pytest.mark.e2e
+@pytest.mark.gpu_1
+@pytest.mark.trtllm_marker
+@pytest.mark.slow
+def test_request_cancellation_trtllm_aggregated(request, runtime_services):
+    """
+    End-to-end test for request cancellation functionality with TRTLLM aggregated backend.
+
+    This test verifies that when a request is cancelled by the client,
+    the system properly handles the cancellation and cleans up resources
+    on the TRTLLM backend side. Tests three scenarios:
+    1. Completion request
+    2. Chat completion request (non-streaming)
+    3. Chat completion request (streaming)
+    """
+
+    trtllm_dir = os.environ.get("TRTLLM_DIR", "/workspace/components/backends/trtllm")
+
+    # Configuration for TRTLLM aggregated cancellation test
+    trtllm_config = TRTLLMConfig(
+        name="aggregated_cancellation_test",
+        directory=trtllm_dir,
+        script_name="agg.sh",
```

Comment:
Are you using the frontend process from the "agg.sh" and "disagg.sh"? Then, you shouldn't need the `class DynamoFrontendProcess(...)` on top?